bin/lint-migrations-file/src/change_set/strict.clj

```diff
@@ -28,7 +28,69 @@
       (apply distinct? (mapcat #(str/split (-> % :sql :dbms) #",")
                                changes))))))
 
+(def change-types-requiring-rollback
+  ;; this list was generated with a little grep and awk from the docs here:
+  ;; https://docs.liquibase.com/workflows/liquibase-community/liquibase-auto-rollback.html
+  #{:addAutoIncrement
+    :alterSequence
+    :createFunction
+    :createPackage
+    :createPackageBody
+    :createProcedure
+    :createTrigger
+    :customChange
+    :delete
+    :dropAllForeignKeyConstraints
+    :dropCheckConstraint
+    :dropColumn
+    :dropDefaultValue
+    :dropForeignKeyConstraint
+    :dropFunction
+    :dropIndexNot
+    :dropPackage
+    :dropPackageBody
+    :dropPrimaryKey
+    :dropProcedure
+    :dropSequence
+    :dropSynonym
+    :dropTable
+    :dropTrigger
+    :dropUniqueConstraint
+    :dropView
+    :empty
+    :executeCommand
+    :insert
+    :loadData
+    :loadUpdateData
+    :markUnused
+    :mergeColumns
+    :modifyDataType
+    :output
+    :setColumnRemarks
+    :setTableRemarks
+    :sql
+    :sqlFile
+    :stop
+    :update})
```

Comment:
I would rather error on the side of whitelisting the ones that have auto-rollback as opposed to blacklisting ones that do not... maybe at some point in the future Liquibase adds a new change type that doesn't support auto-rollback and we use it and break reverse migrations as a result. Might be far-fetched but the linter is supposed to save us from doing something dumb so I'd rather error on the side of caution

Reply:
That's an excellent point, thanks. Will adjust.